include/matrix3x4SIMD.h

```diff
@@ -389,25 +408,28 @@ struct matrix3x4SIMD
 		const __m128 mask1110 = _mm_castsi128_ps(_mm_setr_epi32(0xffffffff, 0xffffffff, 0xffffffff, 0));
 		vectorSIMDf col3;
 
-		vectorSIMDf tmp = (-vectorSIMDf(r0) & mask1110) * center;
-		__m128 ttmp = tmp.getAsRegister();
-		ttmp = _mm_hadd_ps(ttmp, ttmp);
-		ttmp = _mm_hadd_ps(ttmp, ttmp);
-		col3.x = vectorSIMDf(ttmp).x;
-
-		tmp = (-vectorSIMDf(r1) & mask1110) * center;
-		ttmp = tmp.getAsRegister();
-		ttmp = _mm_hadd_ps(ttmp, ttmp);
-		ttmp = _mm_hadd_ps(ttmp, ttmp);
-		col3.y = vectorSIMDf(ttmp).x;
-
-		tmp = (-vectorSIMDf(r2) & mask1110) * center;
-		ttmp = tmp.getAsRegister();
-		ttmp = _mm_hadd_ps(ttmp, ttmp);
-		ttmp = _mm_hadd_ps(ttmp, ttmp);
-		col3.z = vectorSIMDf(ttmp).x;
-
-		col3 += (center - translation) & mask1110;
+		//vectorSIMDf tmp = (-vectorSIMDf(r0) & mask1110) * _center;
+		//__m128 ttmp = tmp.getAsRegister();
+		//ttmp = _mm_hadd_ps(ttmp, ttmp);
+		//ttmp = _mm_hadd_ps(ttmp, ttmp);
+		//col3.x = vectorSIMDf(ttmp).x;
+
+		//tmp = (-vectorSIMDf(r1) & mask1110) * _center;
+		//ttmp = tmp.getAsRegister();
+		//ttmp = _mm_hadd_ps(ttmp, ttmp);
+		//ttmp = _mm_hadd_ps(ttmp, ttmp);
+		//col3.y = vectorSIMDf(ttmp).x;
+
+		//tmp = (-vectorSIMDf(r2) & mask1110) * _center;
+		//ttmp = tmp.getAsRegister();
+		//ttmp = _mm_hadd_ps(ttmp, ttmp);
+		//ttmp = _mm_hadd_ps(ttmp, ttmp);
+		//col3.z = vectorSIMDf(ttmp).x;
+		
+		//result slightly (but significantly) differ from above.. why?
```

Comment:
there's no fast way to insert a custom translation into a row matrix, so we must do it simply in non-SIMD fashion (you could do two transposes but thats simply slower).

there's no need for a transpose
so  multiply all rows by center to obtain `r_i = ri*_center`
then use 3 hadds to reduce the 3 rows (and a random register) into a single 4x1 vector `col3`
then get `finalCol3 = _center-_translation-col3`
you can then save the 3 out of 4 components of finalCol3 to elements 3,7,11 of the matrix with regular C memory ops